hazelcast/src/main/java/com/hazelcast/kubernetes/KubernetesClient.java

```diff
@@ -376,11 +401,15 @@ private static String extractZone(JsonObject nodeJson) {
         for (JsonValue item : toJsonArray(endpointsListJson.get("items"))) {
             String service = toString(item.asObject().get("metadata").asObject().get("name"));
             List<Endpoint> endpoints = parseEndpoints(item);
+
             // Service must point to exactly one endpoint address, otherwise the public IP would be ambiguous.
             if (endpoints.size() == 1) {
                 EndpointAddress address = endpoints.get(0).getPrivateAddress();
-                if (left.contains(address)) {
-                    result.put(address, service);
+                if (privateAddresses.contains(address)) {
+                    // If multiple services match the pod, then match service and pod names
+                    if (!result.containsKey(address) || service.equals(extractTargetRefName(item))) {
```

Comment:
Just to make sure I understand... if a `servicePerPodLabel` has been provided, the load-balancer should not be part of the services, even if it points to only 1 pod, and therefore we should not even run this code - good. If `servicePerPodLabel` has *not* been provided, and the load-balancer points to only 1 pod, we will end up running this code. In which case we want to add the address (to `result`) only if the service name `service` matches the target name. But I'm not too sure about `!result.containsKey(address)`: it means that if you consider the load-balancing service first, you will add it (since the address hasn't been added already) regardless of the name?

Reply:
> Just to make sure I understand... if a `servicePerPodLabel` has been provided, the load-balancer should not be part of the services, even if it points to only 1 pod, and therefore we should not even run this code - good. 

Yes, I guess if you provide the explicit configuration, then load-balancer should not be part of the external services even if it fits.

> If `servicePerPodLabel` has _not_ been provided, and the load-balancer points to only 1 pod, we will end up running this code. In which case we want to add the address (to `result`) only if the service name `service` matches the target name. But I'm not too sure about `!result.containsKey(address)`: it means that if you consider the load-balancing service first, you will add it (since the address hasn't been added already) regardless of the name?

I guess that if you don't provide `service-per-pod` label, then we try to automatically find the "right" services. So any service pointing to only one pod will do, but the priority takes the service with the same name. But if there is no service with the same name as the pod, then we proceed with any service that points to one pod only. This way we're backward-compatible.

Reply:
I need to run tests. It looks like if the first address that you consider is the discovery service, because it's not in `result` already you're going to add it regardless of names, and then the original issue (discovery service being reported as a member service) is not fixed. But I may be wrong.

Reply:
Discussed offline. It should be fine as it is.

Reply:
Tested OK.